### DISCUSSION TITLE
Random seed

### DISCUSSION THREAD

**Discussion by MarcoMarinucci** [galacticusorg/galacticus#334] [Q&A] [answered]
Hi, I 'm using Galacticus and I need to run it assuming 3 different values of the parameter sigma8 to take a numerical derivative over this parameter and I want to check that the results that I have are not just noise. 
First question: when I run Gal., for instance, with sigma8= 0.85 and sigma8 = 0.83 does it use the same random seed for the branching event occurrence?
Second question: if yes, where can I change the value of the random seed? if not, can I fix the same one?

**Comment by abensonca** [comment, marked as answer]
1. Yes, the random number seed is fixed (either by what you set it to in the parameter file, or to the default value) for each run.
2. To change the random number seed you'll want to change the parameter "`randomNumberGeberator`" (or add it if it's not present),, e.g.:
```
  <!-- Random number generation -->
  <randomNumberGenerator value="GSL">
    <seed value="8122"/>
  </randomNumberGenerator>
```
You can change the value of `seed` to some other integer to generate a different random realization of trees.

A couple of additional points:
1. Each tree is actually given it's own random number generator seeded with a value of `seed`+`treeIndex`, where `seed` is the value given in the parameter above and `treeIndex` is the unique index given to the tree. Usually this doesn't matter, but if you're generating _N_ trees and want to generate a new set with completely different initial random number generator seeds you should change `seed` by at least _N_.
2. If you keep `seed` fixed and change σ_8, the trees you get will be similar but not identical. The sequence of random numbers used will be the same, so the initial structure of the branches will be similar, but eventually the difference in σ_8 will lead to some branching that occurred in one set not occurring in the other, and after that the trees will differ. Again, usually this doesn't matter too much, but if you want completely independent trees then change `seed` by at least _N_.
